tests/tests.rs

```diff
@@ -0,0 +1,109 @@
+extern crate froggy;
+
+use froggy::Storage;
+
+#[test]
+fn change_by_pointer() {
+    let storage = Storage::new();
+    {
+        let mut s = storage.write();
+        for &v in [5 as i32, 7, 4, 6, 7].iter() {
+            s.create(v);
+        }
+    }
+    assert_eq!(storage.read().iter().count(), 5);
+    assert_eq!(*storage.read().iter().nth(0).unwrap(), 5);
+    assert_eq!(*storage.read().iter().nth(1).unwrap(), 7);
+    let ptr = {
+        let r = storage.read();
+        let item = r.iter().find(|v| **v == 4).unwrap();
+        r.pin(&item)
+    };
+    let mut w = storage.write();
+    w[&ptr] = 350 as i32;
+    assert_eq!(w[&ptr], 350);
+}
+
+#[test]
+#[should_panic]
+#[allow(unused_variables)]
+fn write_and_read() {
+    let storage: Storage<i32> = Storage::new();
+    let write_lock = storage.write();
+    let read_lock = storage.read();
```

Comment:
hah, I suppose we panic because we unwrap the `PoisonError` from `RwLock`.
This makes me think if we need `try_read`/ `try_write` variants